fetch.bs

```diff
@@ -4648,21 +4648,102 @@ steps:
 
      <li><p>Let <var>body</var> be <var>bodyWithType</var>'s <a for="body with type">body</a>.
 
-     <li><p>Let <var>length</var> be <var>body</var>'s <a for=body>length</a>,
+     <li><p>Let <var>fullLength</var> be <var>body</var>'s <a for=body>length</a>,
      <a lt="serialize an integer">serialized</a> and <a>isomorphic encoded</a>.
 
      <li><p>Let <var>type</var> be <var>bodyWithType</var>'s <a for="body with type">type</a> if it
      is non-null; otherwise the empty byte sequence.
 
-     <li><p>Return a new <a for=/>response</a> whose <a for=response>status message</a> is
-     `<code>OK</code>`, <a for=response>header list</a> is « (`<code>Content-Length</code>`,
-     <var>length</var>), (`<code>Content-Type</code>`, <var>type</var>) », and
-     <a for=response>body</a> is <var>body</var>.
+     <li>
+      <p>If <var>response</var>'s <a for=response>range-requested flag</a> is unset, run the
+      following steps:
+
+      <ol>
+       <li><p>Set <var>response</var>'s <a for=response>status message</a> to `<code>OK</code>`.
+
+       <li><p>Set <var>response</var>'s <a for=response>body</a> to <var>body</var>.
+
+       <li><p>Set <var>response</var>'s <a for=response>header list</a> to «
+       (`<code>Content-Length</code>`, <var>fullLength</var>), (`<code>Content-Type</code>`,
+       <var>type</var>) ».
+      </ol>
+
+     <li>
+      <p>Otherwise, if <var>response</var>'s <a for=response>range-requested flag</a> is set, run the
+      following steps:
+
+      <ol>
+       <li><p>Let <var>rangeHeader</var> be the result of <a>extracting header list values</a> given
+       `<code>Range</code>` and <var>request</var>'s <a for=response>header list</a>.
+
+       <li><p>Let <var>rangeValue</var> be the result of <a>parsing a single range header value</a>
+       given <var>rangeHeader</var>.
+
+       <li><p>If <var>rangeValue</var> is failure, then return a <a>network error</a>.
+
+       <li><p>Let <var>sliceEndRange</var> be null.
+
+       <li>
+        <p>If <var>rangeValue</var>[1] is non-null, let <var>sliceEndRange</var> be
+        <var>rangeValue</var>[1] incremented by one.
+
+        <p class="note">A range header denotes an inclusive byte range, while the <a>slice blob</a>
+        algorithm input range does not. To use the <a>slice blob</a> algorithm, we must increment
+        the parsed range header end value.
+
+       <li><p>Let <var>slicedBlob</var> be the result of invoking <a>slice blob</a> given
+       <var>blobURLEntry</var>'s <a for="blob URL entry">object</a>, <var>rangeValue</var>[0],
+       <var>sliceEndRange</var>, and <var>type</var>.
+
+       <li><p>Let <var>slicedBodyWithType</var> be the result of
+       <a for=BodyInit>safely extracting</a> <var>slicedBlob</var>.
+
+       <li><p>Set <var>response</var>'s <a for=response>body</a> to <var>slicedBodyWithType</var>'s
+       <a for="body with type">body</a>.
+
+       <li><p>Let <var>contentRange</var> be `<code>bytes </code>`.
```

Comment:
I didn't move `contentRange` into another algorithm, but this should definitely be done. I'm currently planning on doing this in a follow up issue, but I can do it here if that is preferred.

Reply:
I think that's fine, but let's make sure it's filed and referenced from an HTML comment?

Reply:
:+1: Added a HTML comment and plan to open a issue when merged. I can open one now if that is the preferred workflow.

Reply:
Yeah please open it now so we can link it from the comment.

Reply:
Created https://github.com/whatwg/fetch/issues/1552.